dotnet/src/SemanticKernel/Memory/MemoryRecord.cs

```diff
@@ -1,25 +1,43 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
 
+using System;
 using System.Text.Json;
+using System.Text.Json.Serialization;
 using Microsoft.SemanticKernel.AI.Embeddings;
 
 namespace Microsoft.SemanticKernel.Memory;
 
 /// <summary>
 /// IMPORTANT: this is a storage schema. Changing the fields will invalidate existing metadata stored in persistent vector DBs.
 /// </summary>
-public class MemoryRecord : IEmbeddingWithMetadata<float>
+public class MemoryRecord : DataEntryBase
 {
     /// <summary>
     /// Source content embeddings.
     /// </summary>
+    [JsonPropertyName("embedding")]
     public Embedding<float> Embedding { get; }
 
     /// <summary>
     /// Metadata associated with a Semantic Kernel memory.
     /// </summary>
+    [JsonPropertyName("metadata")]
     public MemoryRecordMetadata Metadata { get; }
 
+    /// <summary>
+    /// Constructor, use <see cref="ReferenceRecord"/> or <see cref="LocalRecord"/>
+    /// </summary>
```

Comment:
No description of the constructor's parameters?

Reply:
added